src/engine/pieces/piece.ts

```diff
@@ -13,8 +17,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
```

Comment:
I don't think you are using this so it can be removed

Reply:
Fixed